ts/javascript/mod.ts

```diff
@@ -52,9 +58,13 @@ export function mkNpmProject(args: {
   node_modules: Package;
 } {
   const { pkgs, nodejs } = fromNodeVersion(args.nodeVersion);
+  const npmlock2nixRepo = nixFlakeDep("npmlock2nix-repo", {
+    url: "github:nix-community/npmlock2nix?rev=9197bbf397d76059a76310523d45df10d2e4ca81",
+    flake: false,
+  });
```

Comment:
When ignoring the internals of the `nix.ts` module, the API of `nixFlakeDep` is a bit weird. Why do I pass in the name as a parameter and then `url` and `flake` as an object? Maybe instead have it be this:

```suggestion
  const npmlock2nixRepo = nixFlakeDep(
    "npmlock2nix-repo",
    "github:nix-community/npmlock2nix?rev=9197bbf397d76059a76310523d45df10d2e4ca81",
    flake: false,
  );
```

I.e.: `nixFlakeDep` should have type `(name: string, url: string, flake: boolean = false): NixExpression`.

Reply:
Are there other potential options that go on a flake input other than `url` and `flake`? I kind of assumed there might be others which is why I wanted the type, but I couldn't find documentation on this. If not maybe your proposal is best, but also the `false` would not be annotated as it is in your example which makes it less readable. But maybe that's ok?

Reply:
Thinking about this more, while it is nice to re-use the type (that I would hope eventually includes all the valid values of a flake input), the fact that some of these keys are optional are a bit annoying (see `flakeDepEq` and it's equality check of `flake` parameter). Maybe we should have separate types for the input that is more lax than the type in the AST? Would like to hear your thoughts.

Reply:
(Talked about this offline.) I think it's fair that the `nix` module exposes an API that is very similar to what `nix` flakes look like. So I think the current version is best for now.